PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp

```diff
@@ -3052,14 +3075,11 @@ int Progress::thread()
 
 int Progress::createProgressWindow()
 {
-	DWORD styleEx = WS_EX_OVERLAPPEDWINDOW;
-	if (_hOwner)
-		styleEx |= WS_EX_TOOLWINDOW;
-
-	_hwnd = ::CreateWindowEx(styleEx,
+	_hwnd = ::CreateWindowEx(
+		WS_EX_TOOLWINDOW | WS_EX_OVERLAPPEDWINDOW | WS_EX_TOPMOST,
 		cClassName, _header, WS_POPUP | WS_CAPTION,
 		CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT,
-		_hOwner, NULL, _hInst, (LPVOID)this);

```

Comment:
> Actually I did add this.
That's actually kinda funny.

>  I need to pass my class instance to the static window procedure. The way to do it is via CreateWindowEx LPVOID param

I was curious, because I know that the standard guarantees that any pointer (to an object) can implicitly convert to an (identically `const`/`volatile`) `void*`, no explicit `(void*)` needed.

So I went ahead, and, on a local build, changed:

```
_hOwner, NULL, _hInst, (LPVOID)this);
```

to: 

```
_hOwner, NULL, _hInst, this);
```

...and everything compiled without issue. No need for the explicit cast - if you ever need to explicitly cast something to `LPVOID` (or `PVOID`, because we're not writing code for 16-bit systems), and it's not of different `const`/`volatile` qualification....well, uh, try without the cast!

I remember something like this as the culprit in a recent Kernel-Mode vulnerability in Windows, but can't find the original article.